hack/generator/pkg/functions/pivot_conversion_function.go

```diff
@@ -0,0 +1,173 @@
+/*
+ * Copyright (c) Microsoft Corporation.
+ * Licensed under the MIT license.
+ */
+
+package functions
+
+import (
+	"fmt"
+
+	"github.com/dave/dst"
+
+	"github.com/Azure/azure-service-operator/hack/generator/pkg/astbuilder"
+	"github.com/Azure/azure-service-operator/hack/generator/pkg/astmodel"
+	"github.com/Azure/azure-service-operator/hack/generator/pkg/conversions"
+)
+
+// PivotConversionFunction implements a pivot that's used when a conversion reaches a hub type but hasn't yet found the
+// innermost required conversion needed.
+//
+// We use this for Spec types, implementing genruntime.ConvertibleSpec, and for Status types, implementing
+// genruntime.ConvertibleStatus.
+//
+// If the two instances involved in conversion are not in the same "spoke" leading to this "hub" version, we'll pivot to
+// the reverse conversion, starting at the far end of that "spoke":
+//
+// func (s <hubtype>) ConvertFrom(instance <interfaceType>>) error {
+//     return instance.ConvertTo(s)
+// }
+//
+type PivotConversionFunction struct {
+	// nameFrom is the name for this function when converting FROM a provided instance
+	nameFrom string
+	// nameTo is the name for this function when converting TO a provided instance
+	nameTo string
+	// parameterType is common interface type for our parameter
+	parameterType astmodel.TypeName
+	// direction is the direction of conversion - either FROM the supplied instance, or TO the supplied instance
+	// We initialize this from the supplied property conversion function to guarantee consistency that the function
+	// we generate is using that function correctly
+	direction conversions.Direction
+	// idFactory is a reference to an identifier factory used for creating Go identifiers
+	idFactory astmodel.IdentifierFactory
+}
+
+// Ensure we properly implement the function interface
+var _ astmodel.Function = &PivotConversionFunction{}
+
+// NewSpecPivotConversionFunction creates a pivot conversion function that works to convert between two Spec types
+// implementing the interface genruntime.ConvertibleSpec.
+// hubType is the TypeName of our hub type
```

Comment:
`hubType` isn't mentioned in this function - is this from a previous version?

Reply:
Yup. Fixed up the doccomment.